rules-reviewed/jboss-eap/websphere/WebSphereMQ.windup.xml

```diff
@@ -0,0 +1,1363 @@
+            <addon id="org.jboss.windup.rules,windup-rules-java,2.2.0.Final"/>
+        </dependencies>
+        <sourceTechnology id="websphere" versionRange="(10,12]"/>
+        <sourceTechnology id="websphere-mq" versionRange="(10,12]"/>
+        <targetTechnology id="eap" versionRange="[6,)"/>
+        <targetTechnology id="jms" versionRange="[6,)"/>
+        <tag>reviewed-2015-06-05</tag>
+        <tag>messaging</tag>
+        <tag>mq</tag>
+        <tag>mqe</tag>
+        <tag>ibm</tag>
+        <tag>jms</tag>
+    </metadata>
+
+
+    <rules>
+        <!-- MQe*Adapter -->
+        <rule id="websphere-mq-8jyqn">
+            <when>
+                <javaclass references="com.ibm.mqe.adapters.MQe{*}Adapter">
+                    <location>INHERITANCE</location>
+                </javaclass>
+            </when>
+            <perform>
+                <classification title="WebSphere MQ JMS adapter." effort="4">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/adapters/package-summary.html"/>
+                </classification>
+                <hint>
+                    <message>
+                        WebSphere Startup Service can be replaced with EJB 3.1 Startup Bean,
+                        using `@Startup @Singleton` class and `@PostConstruct` method.
+                    </message>
+                </hint>
+            </perform>
+        </rule>
+
+        <!-- MQe*AdminMsg -->
+        <rule id="websphere-mq-32f7a">
+            <when>
+                <javaclass references="com.ibm.mqe.administration.MQe{*}AdminMsg">
+                    <location>INHERITANCE</location>
+                </javaclass>
+            </when>
+            <perform>
+                <classification title="WebSphere MQ JMS administration API" description="Provides classes used to administer and monitor a Websphere MQ Everyplace queue manager." effort="4">
+                    <link description="Javadoc" href="http://www-01.ibm.com/support/knowledgecenter/SSVHEW_6.2.0/com.ibm.mqe.javadoc/java/com/ibm/mqe/adapters/package-summary.html"/>
+                </classification>
+                <hint>
+                    <message>In JBoss EAP 6, the JMS destinations can be controlled with JBoss CLI.</message>
+                    <link description="JBoss CLI" href="https://access.redhat.com/documentation/en-US/JBoss_Enterprise_Application_Platform/6.4/html-single/Administration_and_Configuration_Guide/index.html#Configure_the_JMS_Server1"/>
+                </hint>
+            </perform>
+        </rule>
+
+
+
+        <!-- WebSphere JMS MQe -->
+
+        <!--
+            Properties env = new Properties();
+            env.put(
+                Context.INITIAL_CONTEXT_FACTORY,
+                "com.ibm.mq.jms.context.WMQInitialContextFactory");
+        -->
+        <rule id="websphere-mq-r322f">
+            <when>
+                <!-- WINDUP-615
```

Comment:
I think &lt;or&gt; works, but using a parameter with a "when" conditional would be better, IMO.